warnonfailure/close.go

```diff
@@ -0,0 +1,34 @@
+// Package warnonfailure contains custom versions of common functions/methods
+// that always log a warning in case of error.
+package warnonfailure
+
+import (
+	"io"
+	"log"
+)
+
+var logPrintf = log.Printf
+
+// Close wraps a resource.Close() call and logs the error, if any, adding a
+// custom message.
+//
+// This function is intended to be used as a better alternative to completely
+// ignoring an error in cases where there isn't any obvious reason to handle it
+// explicitly.
+//
+// Its allows to turn ugly error-logging code such as:
+//    defer func() {
+//        err := resource.Close()
+//        if err != nil {
+//            log.Printf("Warning: ignoring error (%v)", err)
+//        }
+//    }()
+// into a simplified pattern of:
+//    defer Close(resource, "Warning: ignoring error")
+func Close(c io.Closer, msg string) error {
+	err := c.Close()
+	if err != nil {
+		logPrintf("%s (%s)\n", msg, err)
```

Comment:
What is the purpose of logPrintf rather than log.Printf?

Reply:
Being able to swap the implementation with `fmt.Printf` in unit tests, since `log.Printf` adds some bits that make matching the exact output against a known value slightly harder.